unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2544,23 +2567,141 @@ public void InitialRandomSpawn(ServerAction action)
             actionFinished(true, reachablePositions);
         }
 
-        public void GetShortestPath(ServerAction action) {
+        public void ObjectNavExpertAction(ServerAction action) {
             SimObjPhysics sop = getSimObjectFromTypeOrId(action);
-            if (sop == null) {
+            var path = getShortestPath(sop, true);
+            if (path.status == UnityEngine.AI.NavMeshPathStatus.PathComplete) {
+
+                int parts = (int) Math.Round(360f / rotateStepDegrees);
+                if (Math.Abs((parts * 1.0f) - 360f / rotateStepDegrees) > 1e-5) {
+                    errorMessage = "Invalid rotate step degrees for agent, must divide 360 without a remainder.";
+                    actionFinished(false);
```

Comment:
Add a return statement after the action finished so the method stops executing.

Reply:
Good catch! Done.